### PR TITLE
Migrating Ballerina Kafka module to a separate Repo

### LINKED ISSUE #2: Migrate Kafka Module to a Separate Repository
**Description:**
$subject

### LINKED ISSUE #3: Inconsistent naming in Kafka ProducerConfiguration record
**Description:**
The following fields in `kafka:ProducerConfiguration` record are fields to specify a time period / time interval. As per the convention, these fields should be suffixed with `TimeInMillis`.

*  `maxBlock`
* `linger`

This is a small change, however it is a breaking change for the users who uses these fields. Hence this needs to be done in a Kafka module minor / major release.

### LINKED ISSUE #5: Separate securityProtocol from SSLProtocol record.
**Description:**
The configuration field securityProtocol is used for both authentication and authorization. When the Kafka SSL is implemented, the `securityProtocol` value is used inside the `kafka:Protocols` record, which is used inside the `kafka:SecureSocket` record.

But this field should be separated from here, since it is common for SSL authentication and SASL authorization. 

Currently, we have this same field inside `kafka:Protocols` record, which is used in `kafka:SecureSocket` to authenticate, and also `kafka:AuthenticationConfiguration` which is used to authenticate.

One can argue that we can keep both the fields inside both the records, but then there can be issues with both the authentication and the authorization is used simultaneously.

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose
To migrate Ballerina Kafka module to a separate repository, from Ballerina-Lang repository.
This is a part of Ballerina standard library migration effort.

Fixes #2 
Fixes #3 
Fixes #5

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+org.gradle.caching=true
+group=org.ballerinalang
+version=2.0.0-SNAPSHOT
+kafkaVersion=2.3.1
+ballerinaLangVersion=2.0.0-Preview3
```
Comment (Bhashinee) [inline gradle.properties:5]:
```suggestion
ballerinaLangVersion=2.0.0-Preview4-SNAPSHOT
```

File: kafka-ballerina/Ballerina.toml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+[project]
+org-name = "ballerinax"
+version = "2.0.0"
```
Comment (Bhashinee) [inline kafka-ballerina/Ballerina.toml:3]:
```suggestion
version = "@toml.version@"
```

File: kafka-ballerina/Ballerina.toml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+[platform]
+target = "java8"
+
+    [[platform.libraries]]
+    artifactId = "kafka"
+    version = "2.0.0-SNAPSHOT"
```
Comment (Bhashinee) [inline kafka-ballerina/Ballerina.toml:10]:
```suggestion
    version = "@project.version@"
```

File: kafka-ballerina/Ballerina.toml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    path = "../kafka-native/build/libs/kafka-native-2.0.0-SNAPSHOT.jar"
```
Comment (Bhashinee) [inline kafka-ballerina/Ballerina.toml:11]:
```suggestion
    path = "../kafka-native/build/libs/kafka-native-@project.version@.jar"
```

File: kafka-ballerina/src/kafka/tests/test_utils/kafka_server.bal [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import ballerina/system;
```
Comment (kalaiyarasiganeshalingam) [inline kafka-ballerina/src/kafka/tests/test_utils/kafka_server.bal:1]: Can we add the license header?

Comment (ThisaruGuruge) [reply kafka-ballerina/src/kafka/tests/test_utils/kafka_server.bal:17]: Fixed with a2a091b71c90d31b58299a6ee6e51d4dec82081e

File: kafka-ballerina/Ballerina.toml (hunk 2 above)
Comment (aashikam) [inline kafka-ballerina/Ballerina.toml:3]:
```suggestion
version = "@toml.version@"
```

File: kafka-ballerina/Ballerina.toml (hunk 3 above)
Comment (aashikam) [inline kafka-ballerina/Ballerina.toml:10]:
```suggestion
    version = "@project.version@"
```

File: kafka-ballerina/Ballerina.toml (hunk 4 above)
Comment (aashikam) [inline kafka-ballerina/Ballerina.toml:11]:
```suggestion
    path = "../kafka-native/build/libs/kafka-native-@project.version@.jar"
```

File: kafka-ballerina/build.gradle [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+ */
+
+import org.apache.tools.ant.taskdefs.condition.Os
+
+description = 'Ballerina - Kafka Ballerina'
+
+configurations {
+    jbalTools
+    kafkaBallerinaBirDir
+    kafkaBallerinaJar
```
Comment (aashikam) [inline kafka-ballerina/build.gradle:25]:
```suggestion
```

**Comment by aashikam** [comment]
Trying to build this locally gave me a compilation error. Creating the `consumer` apparently isn't picking up the new organization? @ThisaruGuruge 
Screen Shot 2020-08-28 at 12 22 28 AM

**Comment by ThisaruGuruge** [comment]
> Trying to build this locally gave me a compilation error. Creating the `consumer` apparently isn't picking up the new organization? @ThisaruGuruge
> Screen Shot 2020-08-28 at 12 22 28 AM

@aashikam This is because the compiler plugin is still there in the pack and it is validating the resource function. Until we remove the Kafka module, this will not build correctly. I removed Kafka module from the Ballerina-lang and then built on top of it. It is working.

**Comment by aashikam** [comment]
> > Trying to build this locally gave me a compilation error. Creating the `consumer` apparently isn't picking up the new organization? @ThisaruGuruge
> > Screen Shot 2020-08-28 at 12 22 28 AM
> 
> @aashikam This is because the compiler plugin is still there in the pack and it is validating the resource function. Until we remove the Kafka module, this will not build correctly. I removed Kafka module from the Ballerina-lang and then built on top of it. It is working.

Ah I see.

**Review by aashikam — APPROVED** [review]
LGTM.

**Pull request merged**